src/services/helpers/service.js

```diff
@@ -15,37 +19,58 @@ const checkForToken = (params, app) => {
 module.exports = function setup(app) {
 	class MailService {
 		// POST
-		create({
-			headers,
-			email,
-			replyEmail,
-			subject,
-			content,
-			attachments,
-		}, params) {
-			return checkForToken(params, app)
-				.then(async (user) => {
-					const options = secrets.smtp || secrets.sendmail || {};
-					const transporter = nodemailer.createTransport(options);
-					const mail = {
-						from: process.env.SMTP_SENDER || replyEmail || 'noreply@schul-cloud.org',
-						replyTo: replyEmail || process.env.SMTP_SENDER || 'noreply@schul-cloud.org',
-						headers,
-						to: user ? user.email : email,
-						subject: process.env.NODE_ENV === 'production' ? subject : `[SC-TEST] ${subject}`,
-						html: content.html,
-						text: content.text,
-						attachments,
-					};
-					// send mail with defined transport object in production mode
-					if (process.env.NODE_ENV === 'production' || process.env.FORCE_SEND_EMAIL) {
-						const info = await transporter.sendMail(mail);
-						logger.info(`E-Mail Message sent: ${info.messageId}`);
-						return;
-					}
-					// otherwise print email message object on console
-					logger.debug('E-Mail Message not sent (not in production mode):', mail);
-				}).catch((err) => Promise.reject(err));
+		async create(data, params) {
+			const FORCE_SEND_EMAIL = app.get('FORCE_SEND_EMAIL');
+			const NOTIFICATION_PLATFORM = app.get('NOTIFICATION_PLATFORM');
+
+			if (!NOTIFICATION_PLATFORM || process.env.NOTIFICATION_PLATFORM) {
+				throw new Unavailable('Required Env NOTIFICATION_PLATFORM is not defined');
+			}
+
+			const serviceUrls = app.get('services') || {};
+
+			const user = await checkForToken(params, app);
+
+			const {
+				headers,
+				email,
+				replyEmail,
+				subject,
+				content,
+				// TODO: must be implemented by the mailservice
+				// currently only used by the helpdesk
+				// attachments,
+			} = data;
+
+			const Mail = {
+				to: user ? user.email : email,
+				subject,
+				text: content.text,
+				html: content.html,
+				from: SMTP_SENDER || replyEmail || 'noreply@schul-cloud.org',
+				replyTo: replyEmail || SMTP_SENDER || 'noreply@schul-cloud.org',
+			};
+
+			const requestOptions = {
+				uri: `${serviceUrls.notification}/mails`,
+				method: 'POST',
+				headers: {
+					...headers,
+				},
+				body: {
+					platform: NOTIFICATION_PLATFORM,
```

Comment:
```suggestion
					platformId: NOTIFICATION_PLATFORM,
```
I think here needs to be platformId, just tested it with Postman

![Screen Shot 2020-01-07 at 13 26 04](https://user-images.githubusercontent.com/7567881/71895447-63851700-3151-11ea-8bcf-b97aaef5eebc.png)

Reply:
otherwise it is not setting the redis queue right